### PR TITLE
Add scope information to local variables to show only the current scope variables in the debug hit message

### PR DESCRIPTION
**Pull request by pranavan15**
## Purpose

Currently, debug hit messages include all the local variables available, irrespective of the scope and current execution line (issue https://github.com/ballerina-platform/ballerina-lang/issues/9965). This PR resolves it by adding scope information to local variables at the runtime to show only the current scope variables in the debug hit messages.  This PR also includes the following fixes or modifications.

- Rename `Struct` as `Object`/`Record` in the debug hit message (resoves https://github.com/ballerina-platform/ballerina-lang/issues/9962)
- Add non-public fields of an object to its strigified value, which will be printed in the debug hit message (resolves https://github.com/ballerina-platform/ballerina-lang/issues/9964)
- Modify debug test suite
  a. Modifications to include local/global variable inspections
  b. New test case to check the newly introduced local variable scoping

### CODE REVIEW

File: bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java [hunk 1]
```diff
@@ -282,6 +282,43 @@ public String stringValue() {
         }
     }
 
+    /**
+     * String value ignoring the access restrictions.
+     * Non-public fields of an object will also be stringified.
+     *
+     * @return string value
+     */
+    public String stringValueIgnoreAccessRestriction () {
```
Comment (Kishanthan) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java]: Shall we come up with a better method name here?

File: bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java [hunk 2]
```diff
@@ -282,6 +282,43 @@ public String stringValue() {
...
+        readLock.lock();
+        StringJoiner sj = new StringJoiner(", ", "{", "}");
+        try {
+            switch (type.getTag()) {
+                case TypeTags.OBJECT_TYPE_TAG:
+                    for (BField field : ((BStructureType) this.type).getFields()) {
```
Comment (Kishanthan) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java]: Lets check on whether java 8 streams api can be used to write better code in writing this logic?

File: bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java [hunk 3]
```diff
@@ -282,6 +282,43 @@ public String stringValue() {
...
+                        String fieldName = field.getFieldName();
+                        V fieldVal = get((K) fieldName);
+                        sj.add(fieldName + ":" + getStringValue(fieldVal));
+                    }
+                    break;
+                case TypeTags.JSON_TAG:
+                    return getJSONString();
+                default:
+                    String keySeparator = type.getTag() == TypeTags.MAP_TAG ? "\"" : "";
+                    for (Iterator<Map.Entry<K, V>> i = map.entrySet().iterator(); i.hasNext();) {
```
Comment (Kishanthan) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java]: Same here. Lets check on whether java 8 streams api can be used to write better code in writing this logic?

File: bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java (hunk 1 above)
Comment (pranavan15) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java]: Ack; Renamed the method.

File: bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java (hunk 2 above)
Comment (pranavan15) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java]: Rewrote the logic with Java 8 streams API.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java [hunk 4]
```diff
@@ -3693,4 +3702,26 @@ private void loadStructField(BLangExpression fieldAccessExpr, Operand varRefRegI
...
+    private void setVariableScopeStart(LocalVariableInfo localVarInfo, BLangVariable varNode) {
+        if (varNode.pos != null) {
+            localVarInfo.scopeStartLineNumber = varNode.pos.sLine;
+        }
+    }
+
+    private void setVariableScopeEnd(LocalVariableInfo localVarInfo, BLangVariable varNode) {
+        if ((varNode.parent == null) && (varNode.pos != null)) {
+            localVarInfo.scopeEndLineNumber = varNode.pos.eLine;
+        } else {
```
Comment (lankavitharana) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java]: shall we return here and improve the else part?

Comment (pranavan15) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java]: Ack; Modified as suggested.

**Pull request merged**